smack-core/src/main/java/org/jivesoftware/smack/sasl/gssApi/GssApiManager.java

```diff
@@ -0,0 +1,72 @@
+package org.jivesoftware.smack.sasl.gssApi;
+
+import org.ietf.jgss.*;
+import org.jivesoftware.smack.XMPPConnection;
+
+import java.security.Provider;
+
+public class GssApiManager {
+
+    GSSManager gssManager;
+    GSSName userName = null;
+    GSSName serverName = null;
+    GSSCredential userCredential = null;
+    GSSContext clientContext = null;
+
+    Oid objectIdentifiers;
+    byte[] clientToken;
+
+    private static GssApiManager gssApiManager = null;
+
+    public static synchronized GssApiManager getInstanceFor(XMPPConnection connection) {
+        if (gssApiManager == null) {
+            return new GssApiManager(connection);
+        }
+        return gssApiManager;
+    }
+
+    private GssApiManager(XMPPConnection connection){
+        gssManager = GSSManager.getInstance();
+
+        try {
+            userName = gssManager.createName(connection.getUser().asEntityBareJid().asEntityBareJidString(),GSSName.NT_USER_NAME);
```

Comment:
Nit: Missing spaces after comma.
checkstyle will probably let you know of this as well :P